src/app/modules/color-scheme/color-bar/color-bar.component.ts

```diff
@@ -0,0 +1,42 @@
+import { Component, Input, Output, EventEmitter, OnInit } from '@angular/core';
+import { ColorScheme } from '../color-scheme-popup/color-scheme-popup.component';
+
+@Component({
+  selector: 'ccf-color-bar',
+  templateUrl: './color-bar.component.html',
+  styleUrls: ['./color-bar.component.scss']
+})
+export class ColorBarComponent implements OnInit {
```

Comment:
This component is doing too much. It should only be concerned about a single color scheme. It does not need to be aware of its position in the array of schemes. That is the responsibility of the user component.